jac/support/jac-lang.org/docs/learn/tutorial/code/server.jac

```diff
@@ -0,0 +1,109 @@
+import:py from mtllm.llms {Ollama}
+
+glob llm = Ollama(model_name='llama3.1');
+
+import:jac from rag {RagEngine}
+glob rag_engine:RagEngine = RagEngine();
+
+node Session {
+    has id: str;
+    has chat_history: list[dict];
+    has status: int = 1;
+
+    can 'Respond to message using chat_history as context and agent_role as the goal of the agent'
+    llm_chat(
+        message:'current message':str,
+        chat_history: 'chat history':list[dict],
+        agent_role:'role of the agent responding':str,
+        context:'retrieved context from documents':list
+    ) -> 'response':str by llm();
+
+}
+
+walker interact {
+    has message: str;
+    has session_id: str;
+
+    can init_session with `root entry {
+         visit [-->](`?Session)(?id == self.session_id) else {
+            session_node = here ++> Session(id=self.session_id, chat_history=[], status=1);
+            print("Session Node Created");
+
+            visit session_node;
+        }
+    }
+}
+
+enum ChatType {
+    RAG : 'Need to use Retrievable information in specific documents to respond' = "RAG",
+    QA : 'Given context is enough for an answer' = "user_qa"
+}
+
+node Router {
+    can 'route the query to the appropriate task type'
+    classify(message:'query from the user to be routed.':str) -> ChatType by llm(method="Reason", temperature=0.0);
+}
+
+walker infer {
+    has message:str;
+    has chat_history: list[dict];
+
+    can init_router with `root entry {
+        visit [-->](`?Router) else {
```

Comment:
@TharukaCkasthuri Make a modification here that caters for if the router node is already present but all the chat types are not. In the class we asked students to create a new chat type node for themselves and they struggled to add it.